packages/core/http/core-http-router-server-internal/src/versioned_router/core_versioned_route.ts

```diff
@@ -26,40 +22,38 @@ import type {
   RouteSecurity,
   RouteMethod,
   VersionedRouterRoute,
-  PostValidationMetadata,
+  AnyKibanaRequest,
 } from '@kbn/core-http-server';
-import type { Mutable } from 'utility-types';
+import { Request } from '@hapi/hapi';
+import { Logger } from '@kbn/logging';
+import { Mutable } from 'utility-types';
 import type { HandlerResolutionStrategy, Method, Options } from './types';
 
-import { validate } from './validate';
 import {
   isAllowedPublicVersion,
   isValidRouteVersion,
   hasQueryVersion,
   readVersion,
   removeQueryVersion,
 } from './route_version_utils';
-import { getVersionHeader, injectVersionHeader } from '../util';
+import { injectVersionHeader } from '../util';
 import { validRouteSecurity } from '../security_route_config_validator';
 
 import { resolvers } from './handler_resolvers';
 import { prepareVersionedRouteValidation, unwrapVersionedResponseBodyValidation } from './util';
 import type { RequestLike } from './route_version_utils';
-import { Router } from '../router';
+import { RequestHandlerEnhanced, Router } from '../router';
+import { kibanaResponseFactory as responseFactory } from '../response';
+import { validateHapiRequest } from '../route';
+import { RouteValidator } from '../validator';
+import { CoreKibanaRequest } from '../request';
 
 interface InternalVersionedRouteConfig<M extends RouteMethod> extends VersionedRouteConfig<M> {
   isDev: boolean;
   useVersionResolutionStrategyForInternalPaths: Map<string, boolean>;
   defaultHandlerResolutionStrategy: HandlerResolutionStrategy;
 }
 
-// This validation is a pass-through so that we can apply our version-specific validation later
-export const passThroughValidation = {
-  body: schema.nullable(schema.any()),
-  params: schema.nullable(schema.any()),
-  query: schema.nullable(schema.any()),
-};
```

Comment:
This is an important change of this refactor, bc we have an `InternalRouteHandler` that allows for more distinct route vs versioned route implementations we can avoid the "handler in a handler" flow that existed before causing us to run validation twice (once pass through, once for real).